warp/src/js_exports/raygun.rs

```diff
@@ -301,6 +304,137 @@ impl RayGunBox {
+            .await
+            .map_err(|e| e.into())
+    }
+
+    /// Add a recipient to the conversation
+    pub async fn add_recipient(
+        &mut self,
+        conversation_id: String,
+        recipient: String,
+    ) -> Result<(), JsError> {
+        self.inner
+            .add_recipient(
+                Uuid::from_str(&conversation_id).unwrap(),
+                &DID::from_str(&recipient).unwrap_or_default(),
+            )
+            .await
+            .map_err(|e| e.into())
+    }
+
+    /// Remove a recipient from the conversation
+    pub async fn remove_recipient(
+        &mut self,
+        conversation_id: String,
+        recipient: String,
+    ) -> Result<(), JsError> {
+        self.inner
+            .remove_recipient(
+                Uuid::from_str(&conversation_id).unwrap(),
+                &DID::from_str(&recipient).unwrap_or_default(),
+            )
+            .await
+            .map_err(|e| e.into())
+    }
+}
+
+/// impl RayGunAttachment trait
+#[wasm_bindgen]
+impl RayGunBox {
+    /// Send files to a conversation.
+    /// If no files is provided in the array, it will throw an error
+    pub async fn attach(
+        &mut self,
+        conversation_id: String,
+        message_id: Option<String>,
+        locations: Vec<JsValue>,
+        message: Vec<String>,
+    ) -> Result<AttachmentResult, JsError> {
+        self.inner
+            .attach(
+                Uuid::from_str(&conversation_id).unwrap(),
+                message_id.map(|s| Uuid::from_str(&s).unwrap()),
+                locations
+                    .iter()
+                    .map(|v| serde_wasm_bindgen::from_value(v.clone()).unwrap())
+                    .collect(),
+                message,
+            )
+            .await
+            .map_err(|e| e.into())
+            .map(|(id, ok)| AttachmentResult {
+                file: id.to_string(),
+                stream: AsyncIterator::new(Box::pin(
+                    ok.map(|s| serde_wasm_bindgen::to_value(&AttachmentKind::from(s)).unwrap()),
+                )),
+            })
+    }
+
+    /// Downloads a file that been attached to a message
+    /// Note: Must use the filename associated when downloading
+    pub async fn download(
+        &self,
+        conversation_id: String,
+        message_id: String,
+        file: String,
+        path: String,
+    ) -> Result<AsyncIterator, JsError> {
+        self.inner
+            .download(
+                Uuid::from_str(&conversation_id).unwrap(),
+                Uuid::from_str(&message_id).unwrap(),
+                file,
+                PathBuf::from(path),
+            )
+            .await
+            .map_err(|e| e.into())
+            .map(|ok| {
+                AsyncIterator::new(Box::pin(
+                    ok.map(|s| serde_wasm_bindgen::to_value(&Progression::from(s)).unwrap()),
+                ))
+            })
+    }
```

Comment:
We would need to omit/exclude this since this function is specific towards downloading to a specific path on the filesystem and we dont have that type of access on the browser. Instead, use `RayGun::download_stream`.

Reply:
You can but that information isnt passable here and its more of a security thing for the browser.